src/injected/injectedScript.ts

```diff
@@ -360,15 +360,20 @@ export default class InjectedScript {
       const clientRect = element.getBoundingClientRect();
       const rect = { x: clientRect.top, y: clientRect.left, width: clientRect.width, height: clientRect.height };
       const samePosition = lastRect && rect.x === lastRect.x && rect.y === lastRect.y && rect.width === lastRect.width && rect.height === lastRect.height && rect.width > 0 && rect.height > 0;
+      lastRect = rect;
       if (samePosition)
         ++samePositionCounter;
       else
         samePositionCounter = 0;
-      let isDisplayedAndStable = samePositionCounter >= rafCount;
+      const isDisplayedAndStable = samePositionCounter >= rafCount;
+
       const style = element.ownerDocument && element.ownerDocument.defaultView ? element.ownerDocument.defaultView.getComputedStyle(element) : undefined;
-      isDisplayedAndStable = isDisplayedAndStable && (!!style && style.visibility !== 'hidden');
-      lastRect = rect;
-      return !!isDisplayedAndStable;
+      const isVisible = !!style && style.visibility !== 'hidden';
+
+      const elementOrButton = element.closest('button, [role=button]') || element;
+      const isDisabled = ['BUTTON', 'INPUT', 'SELECT'].includes(elementOrButton.nodeName) && elementOrButton.hasAttribute('disabled');
+
+      return isDisplayedAndStable && isVisible && !isDisabled;
```

Comment:
Should we check for disabled before stability because it is cheaper?

Reply:
I think we still have to save `lastRect`, so the difference would be negligible.